tests/integration/clientv3/naming/resolver_test.go

```diff
@@ -113,6 +113,91 @@ func TestEtcdGrpcResolver(t *testing.T) {
 	}
 }
 
+// This test mimics the round robin scenario described in grpc_naming.md doc.
+
+func TestEtcdGrpcResolverRoundRobin(t *testing.T) {
+	integration2.BeforeTest(t)
+
+	s1PayloadBody := []byte{'1'}
+	s1 := grpc_testing.NewDummyStubServer(s1PayloadBody)
+	if err := s1.Start(nil); err != nil {
+		t.Fatal("failed to start dummy grpc server (s1)", err)
+	}
+	defer s1.Stop()
+
+	s2PayloadBody := []byte{'2'}
+	s2 := grpc_testing.NewDummyStubServer(s2PayloadBody)
+	if err := s2.Start(nil); err != nil {
+		t.Fatal("failed to start dummy grpc server (s2)", err)
+	}
+	defer s2.Stop()
+
+	clus := integration2.NewCluster(t, &integration2.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	em, err := endpoints.NewManager(clus.Client(0), "foo")
+	if err != nil {
+		t.Fatal("failed to create EndpointManager", err)
+	}
+
+	e1 := endpoints.Endpoint{Addr: s1.Addr()}
+	e2 := endpoints.Endpoint{Addr: s2.Addr()}
+
+	err = em.AddEndpoint(context.TODO(), "foo/e1", e1)
+	if err != nil {
+		t.Fatal("failed to add foo", err)
+	}
+
+	b, err := resolver.NewBuilder(clus.Client(1))
+	if err != nil {
+		t.Fatal("failed to new resolver builder", err)
+	}
+
+	conn, err := grpc.Dial("etcd:///foo", grpc.WithInsecure(), grpc.WithResolvers(b),
+		grpc.WithDefaultServiceConfig(`{"loadBalancingPolicy":"round_robin"}`))
+	if err != nil {
+		t.Fatal("failed to connect to foo", err)
+	}
+	defer conn.Close()
+
+	c := testpb.NewTestServiceClient(conn)
+	resp, err := c.UnaryCall(context.TODO(), &testpb.SimpleRequest{}, grpc.WaitForReady(true))
+	if err != nil {
+		t.Fatal("failed to invoke rpc to foo (e1)", err)
+	}
+	if resp.GetPayload() == nil || !bytes.Equal(resp.GetPayload().GetBody(), s1PayloadBody) {
+		t.Fatalf("unexpected response from foo (e1): %s", resp.GetPayload().GetBody())
+	}
+
+	em.DeleteEndpoint(context.TODO(), "foo/e1")
+	em.AddEndpoint(context.TODO(), "foo/e2", e2)
+
+	// We use a loop with deadline of 30s to avoid test getting flake
+	// as it's asynchronous for gRPC Client to update underlying connections.
+	maxRetries := 300
+	retryPeriod := 100 * time.Millisecond
+	retries := 0
+	for {
+		time.Sleep(retryPeriod)
+		retries++
+
+		resp, err = c.UnaryCall(context.TODO(), &testpb.SimpleRequest{})
+		if err != nil {
+			if retries < maxRetries {
+				continue
+			}
+			t.Fatal("failed to invoke rpc to foo (e2)", err)
+		}
+		if resp.GetPayload() == nil || !bytes.Equal(resp.GetPayload().GetBody(), s2PayloadBody) {
+			if retries < maxRetries {
+				continue
+			}
+			t.Fatalf("unexpected response from foo (e2): %s", resp.GetPayload().GetBody())
+		}
+		break
+	}
+}
```

Comment:
Could you please verify here that both services got >0 number of requests. 
Probably this requires expanding dummyStubServer to have a counter of number of invocations. 
I think that even each request can return the state of atomic counter in the response.

Reply:
Great suggestion @ptabor.  I've made an attempt at this as well as trying to re-use one test just with parameter for load balancing policy.  Can you please take another look and let me know what you think?